driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -160,11 +160,20 @@ mod solver_input {
     }
 }
 
+#[cfg_attr(test, mockall::automock)]
+trait IoMethods {
```

Comment:
super naming nit: not a huge fan of calling a trait XYMethods. Maybe this can be IoTrait and the concrete implementation FileIo/DefaultIo (theoretically we could add an IPC IO method later on).

Reply:
Personally I slightly prefer `IoMethods` over `IoTrait`, since the latter seams a little redundant and `trait FooTrait` feels weird. I would rather just call it `Io` or `System` (and then `MockSystem` even sounds good :wink:).